### PR TITLE
Basic image gallery

### PR DESCRIPTION
**Pull request by smith-kyle**
***TLDR: While typing up the PR description I decided I'd rather stick with your grid idea. But we should still merge this since it has some good things that we should keep***

It's not pretty but it's a start.

I'm with you in keeping it simple. This is a super simple start. Basically every page has a single link and it just goes in a loop like 
```
Intro -> 2007 -> 2008 -> ... -> 2017 -> Outro -> Intro -> ...
```

Each year page is a grid of images.

I liked your idea of placing all the images in a grid. Since David asked for music I thought it'd make more sense to have a more clear split between the years so that the songs could start and stop when they jump to the next year.

But now I'm thinking that we can get the best of both worlds with your grid idea.

Ok so we arrange the photos like the Netflix homepage where each row is a different year
```
2007-1    2007-2    2007-3    2007-4    2007-5    2007-6 <- not focused
2008-1    2008-2    2008-3    2008-4    2008-5    2008-6 <- focused year
2009-1    2009-2    2009-3    2009-4    2009-5    2009-6 <- not focused
```
And we can do some sort of visual fade-out effect on the rows that aren't in focus. And fade in the music with the visual fade-in of the focused year. We should cut the songs into the most interesting, emotional parts, like the choruses and just do a bunch of cross-fading between them.

Ok so I'm going to totally change the Year component but I still think we should merge this

### CODE REVIEW

**Comment by smith-kyle** [comment]
This PR has changed a lot since I opened it initially. I've changed the design. Here's a gif
dave-10-demo

File: src/components/Years.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+const getFocusedIndex = (windowHeight, scrollY) => R.pipe(
+  () => [
+    scrollY + windowHeight / 2,
+    getRowMargin(windowHeight) + getRowUnfocusedHeight(windowHeight) + getRowHeaderHeight(windowHeight),
+    getRowMargin(windowHeight) + getRowFocusedHeight(windowHeight) + getRowHeaderHeight(windowHeight)
+  ],
+  ([middleY, totalUnfocusedHeight, totalFocusedHeight]) => R.reduce(
+    ([sum, index], _) => middleY < sum || sum === -1
+      ? [-1, index]
```
Comment (smith-kyle) [inline src/components/Years.jsx:16]: please don't judge me
